lib/sqs.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+const assert = require('assert');
+const AWS    = require('aws-sdk');
+
+
+module.exports = class SQS {
+  constructor(config) {
+    this.accessKeyId     = config.accessKeyId;
+    this.secretAccessKey = config.secretAccessKey;
+    this.region          = config.region;
+    this.queueName       = config.queueName;
+  }
+
+  post(messages) {
+    return this.sqsClient()
+      .then(sqsClient => {
+        const entries = messages.map((message, index) => {
+          const entry = {
+            // Result may be unordered, that's why we need to provide an ID
+            // for each message in this batch.
+            Id:          index.toString(),
+            MessageBody: message.body
+          };
+
+          // This is the recommended way to check if a queue is FIFO or not.
+          if (this.queueName.endsWith('.fifo'))
+            entry.MessageGroupId = this.queueName;
+
+          return entry;
+        });
+
+        const sqsPayload = {
+          QueueUrl: this.queueURL,
+          Entries:  entries
+        };
+
+        return sqsClient.sendMessageBatch(sqsPayload).promise();
+      })
+      .then(response => {
+        const successful = response.Successful;
+        const failed     = response.Failed;
+
+        if (failed.length) {
+          const errors = failed.map(error => `${error.Code}: ${error.Message}`);
+          throw new Error(errors.join(', '));
+        }
+
+        // Reconstruct array of job IDs based on the batch ID for this request.
+        const orderedJobIDs = messages.map(function(message, index) {
+          const responseResult = successful.find(msg => msg.Id === index.toString());
+          assert(responseResult);
```

Comment:
There are a few assumptions:

* We queue N messages, we get N results back if everything worked.
* Results have an `Id` attribute, which is a string, that corresponds to the string ID we provided for the batch request.

I agree a proper error message was missing, I'll redo it.